content/technotes/2018-09-03-commonmark.md

```diff
@@ -0,0 +1,327 @@
+
+This note will appear to you, dear reader, as an html page, either here
+on ropensci.org or on R-Bloggers, but I’m writing it as an R Markdown
+document, using Markdown syntax. I’ll knit it to Markdown and then
+Hugo’s Markdown processor,
+[Blackfriday](https://github.com/russross/blackfriday), will transform
+it to html. Elements such as “\# blabla” thus get transformed to
+`<h1 id="blabla">blabla</h1>`. Awesome!
+
+The rendering of Markdown to html or XML can also be used as a way to
+*parse* it, [which is what the `spelling` package does in order to
+identify text
+segments](https://github.com/ropensci/spelling/blob/622fc9cc200b69b3859f272c99c13c575dcb5105/R/parse-markdown.R#L12)
+of R Markdown files, before spell checking them only, not code. I had an
+aha moment when seeing this `spelling` strategy: why did I ever use
+regex to parse Markdown for text analysis?! Transforming it to XML
+first, and then using XPath, would be much cleaner!
+
+As a side-note, realizing how to simplify my old code made me think of
+[Jenny Bryan’s inspiring useR! keynote talk about code
+smells](https://github.com/jennybc/code-smells-and-feels). I asked her
+whether code full of regular expressions instead of dedicated parsing
+tools was a code smell, sadly it doesn’t have a specific name, but she
+confirmed my feeling that *not* using dedicated purpose-built tools
+might mean you’ll end up “re-inventing all of that logic yourself, in
+hacky way.”. If you have code falling under the definition below, maybe
+try to re-factor and if needed [get
+help](https://masalmon.eu/2018/07/22/wheretogethelp/).
+
+<!--html_preserve-->
+<blockquote class="twitter-tweet" data-lang="en">
+<p lang="en" dir="ltr">
+It’s that feeling when you want to do something that sounds simple but
+instead your code is like 10 stack overflow snippets slapped together
+that you could never explain to another human what they do 😰
+<a href="https://t.co/IF53AX6QvC">pic.twitter.com/IF53AX6QvC</a>
+</p>
+— Dr. Alison Hill (@apreshill)
+<a href="https://twitter.com/apreshill/status/1035526182392000514?ref_src=twsrc%5Etfw">31
+d’agost de 2018</a>
+</blockquote>
+<!--/html_preserve-->
+### From Markdown to XML
+
+In this note I’ll use my local fork of rOpenSci’s website source, and
+use all the Markdown sources of blog posts as example data. The chunk
+below is therefore not portable, sorry about that.
+
+``` r
+roblog <- "C:\\Users\\Maelle\\Documents\\ropensci\\roweb2\\content\\blog"
+
+all_posts <- fs::dir_ls(roblog, regexp = "*.md")
+all_posts <- all_posts[all_posts != "_index.md"]
+```
+
+My fork master branch isn’t entirely synced. It has 202 posts.
+
+The code below uses the [`commonmark`
+package](https://github.com/jeroen/commonmark) to render Markdown to
+XML. Commonmark is a Markdown processor like Blackfriday, except it’s
+used by GitHub. The `commonmark` package by Jeroen Ooms offers bindings
+to Commonmark’s C library. Note that my function still has a hacky
```

Comment:
Maybe replace this part: __Commonmark is a Markdown processor like Blackfriday, except it’s used by GitHub. The `commonmark` package by Jeroen Ooms offers 
bindings to Commonmark’s C library.__

With something like this: __commonmark is a [standardized](https://spec.commonmark.org/0.28/) specification for markdown syntax by [John McFarlane](https://johnmacfarlane.net/tools). The `commonmark` R package wraps the official [cmark](https://github.com/commonmark/cmark) library and is used by e.g. Github to render issues and readmes.__

Reply:
Much better, thanks!